WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/editing/OrderEditingViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.woocommerce.android.ui.orders.details.editing
+
+import androidx.lifecycle.SavedStateHandle
+import com.woocommerce.android.model.Order
+import com.woocommerce.android.ui.orders.details.OrderDetailFragmentArgs
+import com.woocommerce.android.viewmodel.ScopedViewModel
+import com.woocommerce.android.viewmodel.navArgs
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class OrderEditingViewModel @Inject constructor(
+    savedState: SavedStateHandle,
+    private val orderEditingRepository: OrderEditingRepository
+) : ScopedViewModel(savedState) {
+    private val navArgs by savedState.navArgs<OrderDetailFragmentArgs>()
+    private val order: Order
+
+    private val orderIdentifier: String
+        get() = navArgs.orderId
+
+    val customerOrderNote: String
+        get() = order.customerNote
+
+    init {
+        order = orderEditingRepository.getOrder(orderIdentifier)
+    }
+
+    fun updateCustomerOrderNote(updatedCustomerOrderNote: String) {
+        launch {
+            orderEditingRepository.updateCustomerOrderNote(order, updatedCustomerOrderNote)
+        }
+    }
```

Comment:
suggestion: In order to avoid ANRs and following the fact that we're looking to do optimistic updates, we could dispatch this launch operation to be done at `IO` context and leave it behind:

```suggestion
       fun updateCustomerOrderNote(updatedCustomerOrderNote: String) {
        launch(dispatchers.io) {
            orderEditingRepository.updateCustomerOrderNote(order, updatedCustomerOrderNote)
        }
    }
```

We would need just to add the `CoroutineDispatchers` to the ViewModel constructor:

```
class OrderEditingViewModel @Inject constructor(
    savedState: SavedStateHandle,
    private val dispatchers: CoroutineDispatchers,
    private val orderEditingRepository: OrderEditingRepository
)
```

WDYT?

Reply:
@ThomazFB Oh, yes - definitely a good call. I made that change in f13eb3b.